src/App.tsx

```diff
@@ -15,71 +17,86 @@ export const goods = [
+  selectedGoods: string,
+};
 
-    <table className="table">
-      <tbody>
-        <tr data-cy="Good">
-          <td>
-            <button
-              data-cy="AddButton"
-              type="button"
-              className="button"
-            >
-              +
-            </button>
-          </td>
+export class App extends React.Component<{}, State> {
+  state = {
+    selectedGoods: 'Jam',
+  };
 
-          <td data-cy="GoodTitle" className="is-vcentered">
-            Dumplings
-          </td>
-        </tr>
+  render() {
+    const { selectedGoods } = this.state;
 
-        <tr data-cy="Good" className="has-background-success-light">
-          <td>
-            <button
-              data-cy="RemoveButton"
-              type="button"
-              className="button is-info"
-            >
-              -
-            </button>
-          </td>
+    return (
+      <main className="section container">
+        {selectedGoods
+          ? (
+            <h1 className="title is-flex is-align-items-center">
+              {`${selectedGoods} is selected`}
+              {/* eslint-disable-next-line jsx-a11y/control-has-associated-label */}
+              <button
+                data-cy="ClearButton"
+                type="button"
+                className="delete ml-3"
+                onClick={() => {
+                  this.setState({ selectedGoods: '' });
+                }}
+              />
+            </h1>
+          )
+          : (<h1 className="title">No goods selected</h1>)}
 
-          <td data-cy="GoodTitle" className="is-vcentered">
-            Jam
-          </td>
-        </tr>
+        <table className="table">
+          <tbody>
+            {goods.map(item => (
+              <tr
+                data-cy="Good"
+                key={item}
+                className={
+                  classNames(
+                    '',
+                    { 'has-background-success-light': selectedGoods === item },
+                  )
+                }
+              >
+                <td>
+                  <button
+                    data-cy={
+                      selectedGoods === item
+                        ? ('RemoveButton')
+                        : ('AddButton')
+                    }
+                    type="button"
+                    className={
 
-        <tr data-cy="Good">
-          <td>
-            <button
-              data-cy="AddButton"
-              type="button"
-              className="button"
-            >
-              +
-            </button>
-          </td>
+                      classNames(
+                        'button',
+                        { 'is-info': selectedGoods === item },
+                      )
+                    }
+                    onClick={() => {
+                      this.setState({ selectedGoods: item });
+                      if (selectedGoods === item) {
+                        this.setState({ selectedGoods: '' });
+                      }
```

Comment:
Same. move this logic to separate function

Reply:
Please, move it to a separate function, as Alexander said